src/sparkx/MultiParticlePtCorrelations.py

```diff
@@ -0,0 +1,500 @@
+
+    def mean_pt_cumulants(self, particle_list_all_events: List[List[Particle]], compute_error: bool=True, delete_fraction: float=0.4, number_samples: int=100, seed: int=42) -> Union[np.ndarray, Tuple[np.ndarray, np.ndarray]]:
+        """
+        Computes the mean transverse momentum cumulants for each order k
+        from Eqs. B9-B16 in Ref. [1].
+
+        Parameters
+        ----------
+        particle_list_all_events : list
+            List of events, where each event is a list of particle objects.
+        compute_error : bool, optional
+            Whether to compute error estimates (default is True).
+        delete_fraction : float, optional
+            Fraction of data to delete for jackknife method (default is 0.4).
+        number_samples : int, optional
+            Number of jackknife samples (default is 100).
+        seed : int, optional
+            Random seed for reproducibility (default is 42).
+
+        Returns
+        -------
+        np.ndarray or tuple
+            Mean transverse momentum cumulants for each order.
+            If compute_error is True, returns a tuple (kappa, kappa_error).
+
+        Raises
+        ------
+        TypeError
+            If delete_fraction is not a float.
+            If number_samples is not an integer.
+            If seed is not an integer.
+            If compute_error is not a boolean.
+        ValueError
+            If delete_fraction is not between 0 and 1.
+            If number_samples is not greater than 0.
+        """
+        if not isinstance(delete_fraction, float):
+            raise TypeError("delete_fraction must be a float")
+        if not 0.0 < delete_fraction < 1.0:
+            raise ValueError("delete_fraction must be between 0 and 1")
+        if not isinstance(number_samples, int):
+            raise TypeError("number_samples must be an integer")
+        if not number_samples > 0:
+            raise ValueError("number_samples must be greater than 0")
+        if not isinstance(seed, int):
+            raise TypeError("seed must be an integer")
+        if not isinstance(compute_error, bool):
+            raise TypeError("compute_error must be a boolean")
+
+
+        self.N_events = []
+        self.D_events = []
+        self._compute_numerator_denominator_all_events(particle_list_all_events)
+        self.N_events = np.array(self.N_events)
+        self.D_events = np.array(self.D_events)
+
+        kappa_store = np.zeros(self.max_order)
+        kappa_store_error = np.zeros(self.max_order)
+        for order in range(self.max_order):
+            dataN = self.N_events[:, :2*(order+1)]
+            dataD = self.D_events[:, :2*(order+1)]
+            # combine the numerator and denominator for each order in data
+            # always alternate the columns of the numerator and denominator
+            combined_data = np.empty((dataN.shape[0], dataN.shape[1] + dataD.shape[1]), dtype=dataN.dtype)
+            combined_data[:, 0::2] = dataN
```

Comment:
I have to say that this confused me a tiny bit in the beginning. I think we can keep it like this, but the next time I would just use two arguments.

Reply:
I agree, it took a few minutes to come up with this.